### DISCUSSION TITLE
Parent Repo under Submodules Pane

### DISCUSSION THREAD

**Discussion by ssbutt** [gitextensions/gitextensions#11867] [Q&A] [answered]
### Environment

- GitExtensions version: v3.5 also v4.2
- GIT version: 2.41.0.windows.3
- OS version: windows
- .NET version:  v4.0.30319

### Issue description

I created a repo using Git Extensions (v4.2). Although I tried the same in v3.5. The submodules pane shows my own repo under Submodules icon, although I do not have any submodules in my repo. The image attached shows this behavior. I do not have any idea why does my repo appears under Submodules. Is it a bug or do i need to change something in the Settings that I should not see orignal repo under submodules?

![CbEBx1ur](https://github.com/user-attachments/assets/6b530dd5-57f3-4688-a4a3-697b63ffe4e8)

I expect that my repo should not appear under Submodules rather only the submodules that will be added to the repo.

### Steps to reproduce

1. Clone any repo
2. Click the Submodules icon, if not already selected, see image below:.
![image](https://github.com/user-attachments/assets/13322ecd-a1ee-4e5a-8381-cc1339c184bb)
3. The repo name appears as a submodules

### Did this work in previous version of GitExtensions?

No, I experienced the same issue with Git Extensions v3.5.

**Comment by mstv** [comment, marked as answer]
I agree that this may be confusing without submodules
The parent repo (and the folder structure) is shown in order to support switching between submodules and the parent repo (by double click a.o.)

![image](https://github.com/user-attachments/assets/6dda1eea-a566-4c7b-9dc3-ec9183e8afac)   ![image](https://github.com/user-attachments/assets/63acbbd6-f59b-4769-b2b5-9d241e5211ff)

(Without submodules, I suggest to just hide the submodules tree using the toolbar button you have already identified.)

Perhaps, the parent repo should be displayed with a different icon, e.g. ![ShellExtensions](https://github.com/user-attachments/assets/b0e1b9b1-088b-4bc9-b57c-d93411525a42)

**Reply by RussKie** [reply]
The main repo is a supermodule, that's why it is shown in this tree.

**Reply by gerhardol** [reply]
To call the sidepanel item something like "Modules" would be confusing too.
Hide the tree when only a superproject is confusing too.

> Perhaps, the parent repo should be displayed with a different icon, e.g. ShellExtensions

A good suggestion

**Comment by trojanobelix** [comment]
![image](https://github.com/user-attachments/assets/5b3c78f0-74dd-4c76-8276-a00b0de878ca)

It may be clearer to display the supermodule as a tree root. And not on the first submodule level

**Reply by ssbutt** [reply]
This display pane would be really helpful to differentiate between supermodule and submodules
